tests/Cluster.py

```diff
@@ -707,7 +710,17 @@ def getNode(self, nodeId=0, exitOnError=True):
         return self.nodes[nodeId]
 
     def getNodes(self):
-        return self.nodes
+        nodes = []
```

Comment:
If the purpose of this convoluted code is to return a copy of the list of nodes, use a one-liner: `return self.nodes[:]`.  This uses slice notation to compactly copy a list.  `self.nodes` should have been set to an empty list in the class's `__init__()` method so the additional checks for existence and emptiness are not required.